### PR TITLE
[RPKG] Remove duplicated compute IOI hash implementations, more cleanup

### PR DESCRIPTION
**Pull request by RDIL**
- 200% reduction in number of duplicate implementations of IOI hash calculation!
- Use modern for loop syntax in some places.
- Mark some things as constants for additional optimization.
- Move dev functions into an `experimental` folder.
- More c-style casts to c++ style casts.

### CODE REVIEW

**Comment by AnthonyFuller** [comment]
Before I review in about 20 minutes. Could you force push to rename the commit to follow the (current) style?

File: rpkg_src/crypto.cpp [hunk 1]
```diff
@@ -1,71 +1,70 @@
...
-void crypto::xor_data(char *input_data, uint32_t input_data_length)
+void crypto::xor_data(char *input_data, const uint32_t input_data_length)
 {
-    unsigned char xor_array[] = {0xDC, 0x45, 0xA6, 0x9C, 0xD3, 0x72, 0x4C, 0xAB};
-    int xor_length = sizeof(xor_array);
+    const unsigned char xor_array[] = {0xDC, 0x45, 0xA6, 0x9C, 0xD3, 0x72, 0x4C, 0xAB};
 
     for (uint64_t i = 0; i < input_data_length; i++)
     {
+        constexpr int xor_length = sizeof xor_array;
```
Comment (AnthonyFuller) [inline rpkg_src/crypto.cpp:9]: Why is this in here? The size of the array will never change so it should be defined outside. I'm sure the compiler would optimise this anyway but why move it in the first place?

**Review by AnthonyFuller — CHANGES_REQUESTED** [review]
Looks alright to me. I do think a lot of things, such as the const typing, would've been done by the compiler anyway but address what has been flagged.

File: rpkg_src/crypto.cpp (hunk 1 above)
Comment (RDIL) [reply rpkg_src/crypto.cpp:9]: Good point. Will revert later.

File: rpkg_src/experimental/dev_diff_rpkg_supermetas.cpp [hunk 2]
```diff
@@ -129,13 +127,13 @@ void dev_function::dev_diff_rpkg_supermetas(std::string& input_path, std::string
 
                     for (uint64_t a = 0; a < rpkgs.at(x).hash.size(); a++)
                     {
-                        std::map<uint64_t, uint64_t>::iterator it = rpkgs.at(y).hash_map.find(rpkgs.at(x).hash.at(a).hash_value);
+                        auto it = rpkgs.at(y).hash_map.find(rpkgs.at(x).hash.at(a).hash_value);
```
[the remark this answers is not in the document]
Comment (RDIL) [reply rpkg_src/experimental/dev_diff_rpkg_supermetas.cpp:130]: Rider suggestion, primarily because it's shorter.

**Pull request merged**